src/components/dashboard/FaceVerification/screens/VerificationScreen.js

```diff
@@ -50,10 +51,24 @@ const FaceVerification = ({ screenProps }) => {
     fireEvent(FV_PROGRESS_ZOOM)
   }, [])
 
-  const retryHandler = useCallback(eventData => {
-    trackAttempt(eventData.reason)
-    fireEvent(FV_TRYAGAIN_ZOOM, eventData)
-  }, [])
+  const retryHandler = useCallback(
+    eventData => {
+      const exception = eventData.reason
+
+      // tracking all lovenes exceptions exceptions except
+      // the last one required to show 'try again lagter'
+      if (attemptsCount < MAX_RETRIES_ALLOWED) {
+        trackAttempt(exception)
+      } else if (!livenessExceptionRef.current) {
+        // storing the last one in the ref, it will be
+        // tracked & reported by the error screen
+        livenessExceptionRef.current = exception
```

Comment:
i dont understand this
the 3rd failed attempt should show the "unrecoverable" error screen.
this still doesnt seem to be the correct logic

Reply:
If we'll do an direct redirect this will break the session￼ in non correct way. This probably will cause context switch exception and we'll be redirected to the error screen twice . Not sure how zoom will do in this case - we have to try

My logic is pretty simple - we tracking retry errors, storing latest in the ref. If user then cancels fv (after 3 retries it's the most probable scenario as user is tired to taking face snapshots) we'll go on error screen when we tracking the last one retry error showing try again later and reporting to amp

If you want the exact redirect on 3rd retry - we could try to do it.

Reply:
ok so lets do the following
can we move the log.error report after 3 failures to the trackAttempts? so after any 3 failures a log.error will be sent? regardless if the "unrecoverable" was shown or not

Reply:
Yes let's try this way . This removes also a few workarounds we did